File: Majority Element (169)/169.py
# Runtime: 32 ms
# Beats 99.36% of Python submissions

class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        element_multiplicity = {}
        
        for num in nums:
            if num not in element_multiplicity:
                element_multiplicity[num] = 1
            else:
                element_multiplicity[num] += 1
                
        return max(element_multiplicity, key=element_multiplicity.get)

File: Majority Element (169)/169.java
// Runtime: 4 ms
// Beats 89.89% of Java submissions
class Solution {
    public int majorityElement(int[] nums) {
        Random rand = new Random();
        
        while (true) {
            int randInt = nums[rand.nextInt(nums.length)];
            
            if (countElements(randInt, nums) > (nums.length / 2)) {
                return randInt;
            }
        }
    }
    
    public static int countElements(int number, int[] list) {
        int count = 0;
        
        for (int i = 0; i < list.length; i++) {
            if (list[i] == number) {
                count++;
            }
        }
        
        return count;
    }
}